include/capstone/capstone.h

```diff
@@ -289,6 +289,36 @@ typedef struct cs_opt_skipdata {
 	void *user_data;
 } cs_opt_skipdata;
 
+/// Provides information about an operand's encoding in the instruction
+typedef struct cs_operand_encoding {
+	/// Specifies how many pieces that form the full operand are encoded in the
+	/// instruction separately. For example if count is 2 it means a few bits of
+	/// this operand are in one location and the rest on another. If it's 0 then
+	/// the operand is NOT encoded anywhere in the instruction.
+	uint8_t operand_pieces_count;
+	/// The bit positions of each piece that form the full operand in order. If
+	/// there is only one piece then there is only one index as well. Likewise
+	/// if there are 4 pieces, there are 4 indexes and so on.
+	uint8_t indexes[8];
+	/// The bit widths of each piece that form the full operand in order. If
+	/// there is only one piece then there is only one size as well. Likewise if
+	/// there are 4 pieces, there are 4 sizes and so on.
+	uint8_t sizes[8];
+} cs_operand_encoding;
+
+/// Provides information about an operand's opcode in the instruction
+typedef struct cs_opcode_encoding {
+	/// Contains all the bits (in order) that form the full opcode.
+	/// Note that each bit is NOT necessarily next to each other in the
+	/// instruction bytes. (see below)
+	uint64_t bits;
+	/// As mentioned above, since the opcode bits may not be next to each other
+	/// this array comes to the rescue by providing the location of each bit
+	/// individually.
+	uint8_t indexes[64];
+	uint8_t
+		bit_count; ///< Specifies the number of bits that form the full opcode.
+} cs_opcode_encoding;
```

Comment:
After the rebase please move it to `Mapping.h`

Reply:
Isn't the intention for the struct to be publicly available? I might be wrong but `Mapping.h` isn't included by the user.

Reply:
You are right. Leave it there.